modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -1333,6 +1333,9 @@ private void onDisconnect(JdbcThinTcpIo cliIo) {
 
             stmts.clear();
         }
+
+        // Clear local metadata cache on disconnect.
+        metaHnd = new JdbcBinaryMetadataHandler();
```

Comment:
It's not enough just to swap links here, because metaHnd mainly used inside a BinaryContext, thus we have to update that link too

Reply:
What dou you think about re-initialize binary context here?
e.g.:

```
        // Clear local metadata cache on disconnect.
        metaHnd = new JdbcBinaryMetadataHandler();
        ctx = createBinaryCtx(metaHnd, marshCtx);

```

Reply:
Well, can't imagine what could goes wrong, so let's do so